### PR TITLE
add minor stop 10281

### PR DESCRIPTION
**Pull request by alisa911**

### CODE REVIEW

File: native/src/binaryRead.h [hunk 1]
```diff
@@ -308,6 +309,23 @@ struct RouteDataObject {
 		return false;
 	}
 
+	void removePointType(int ind, int type) {
+		if (ind < pointTypes.size()) {
+			std::vector<uint32_t> typesArr = pointTypes[ind];
+			for (int i = 0; i < typesArr.size(); i++) {
```
Comment (vshcherb) [inline native/src/binaryRead.h:315]: much much simpler!

https://stackoverflow.com/questions/3385229/c-erase-vector-element-by-value-rather-than-by-position
pseudocode: 

while (find () == true) { vector.erase(iterator); }

File: native/src/binaryRead.h [hunk 2]
```diff
@@ -308,6 +309,23 @@ struct RouteDataObject {
...
+				if (typesArr[i] == type) {
+					std::vector<uint32_t> result;
+					std::copy(std::begin(typesArr), std::end(typesArr), std::begin(result) + i);
```
Comment (PaulStets) [inline native/src/binaryRead.h:318]: Call to copy does not correspond to java's arraycopy(Object src, int srcPos, Object dest, int destPos, int length).
std::copy(begin, end, position in target vector). In other words, destPos in java is 0, but this code will insert starting from i-th position which is incorrect.

File: native/src/binaryRead.h [hunk 3]
```diff
@@ -308,6 +309,23 @@ struct RouteDataObject {
...
+					if (typesArr.size() != i) {
+						std::copy(std::begin(typesArr) + i + 1, std::end(typesArr) - i, std::begin(result) + i);
```
Comment (PaulStets) [inline native/src/binaryRead.h:320]: Same issue here. This behavior needs to be double-checked with Java

File: native/src/routeResultPreparation.cpp [hunk 4]
```diff
@@ -253,6 +253,38 @@ void splitRoadsAndAttachRoadSegments(RoutingContext* ctx, vector<SHARED_PTR<Rout
...
+	int next;
+
+	for (int j = rr->getStartPointIndex(); j != rr->getEndPointIndex(); j = next) {
+		next = plus ? j + 1 : j - 1;
+		auto pointTypes = rr->object->pointTypes(i);
+
+		if (pointTypes != null) {
+			for (int j = 0; j < pointTypes.length; j++) {
+				if (pointTypes[j] == rr->object->region->stopMinor) {
+					stops.add(i);
```
Comment (PaulStets) [inline native/src/routeResultPreparation.cpp:268]: std::vector does not have an ```add()``` method. Please use ```push_back()``` instead.

File: native/src/routeResultPreparation.cpp [hunk 5]
```diff
@@ -253,6 +253,38 @@ void splitRoadsAndAttachRoadSegments(RoutingContext* ctx, vector<SHARED_PTR<Rout
...
+				}
+			}
+		}
+	}
+
+	for (int stop : stops) {
+		std::vector<SHARED_PTR<RouteSegmentResult>> attachedRoutes = rr->getAttachedRoutes(stop);
+		for (SHARED_PTR<RouteSegmentResult> attached : attachedRoutes) {
```
Comment (PaulStets) [inline native/src/routeResultPreparation.cpp:276]: Use iterators instead. Dereferencing uninitialized objects will cause a crash.

File: native/src/routeResultPreparation.cpp [hunk 6]
```diff
@@ -1640,14 +1672,17 @@ void prepareTurnResults(RoutingContext* ctx, vector<SHARED_PTR<RouteSegmentResul
...
-    
-    prepareTurnResults(ctx, result);
-    return result;
+vector<SHARED_PTR<RouteSegmentResult> > prepareResult(RoutingContext* ctx,
+													  vector<SHARED_PTR<RouteSegmentResult> >& result) {
+	combineWayPointsForAreaRouting(ctx, result);
+	validateAllPointsConnected(result);
+	splitRoadsAndAttachRoadSegments(ctx, result);
+	for (int i = 0; i < result.size(); i++) {
+		filterMinorStops(result.get(i));
```
Comment (PaulStets) [inline native/src/routeResultPreparation.cpp:1681]: Method ```get()``` does not exist fro std::vector. Please use the square braces like you would in Java's primitives array.

**Pull request closed without merging**